subseas_flex_fcst/maproom.py

```diff
@@ -99,26 +99,16 @@ def display_relevant_control(variable):
     Output("leadTime","value"),
     Input("startDate","value"),
 )
-def targetStartOptions(startDate):
-    leadsValues = list(CONFIG["leads"].values())
-    leadsKeys = list(CONFIG["leads"])
-    startDate = pd.to_datetime(startDate)
-    optionsDict = {}
-    for idx, x in enumerate(leadsValues):
-        endDate = x+CONFIG["target_period_length"]-1 #used to calculate the target end date (targetEnd = lead + period length - 1)
-        targetStart = startDate + timedelta(days=x)
-        targetEnd = startDate + timedelta(days=endDate)
-        if (targetStart).strftime("%Y") == (targetEnd).strftime("%Y"):
-            if (targetStart).strftime("%b") == (targetEnd).strftime("%b"):
-                targetStartStr = targetStart.strftime("%-d")
-            else:
-                targetStartStr = (targetStart).strftime("%-d %b")
-        else:
-            targetStartStr = (targetStart).strftime("%-d %b %Y")
-        targetEndStr = targetEnd.strftime("%-d %b %Y")
-        dateRange = f"{targetStartStr} - {targetEndStr}"
-        optionsDict.update({leadsKeys[idx]:dateRange})
-    return optionsDict, leadsKeys[0]
+def targetStartOptions(start_date):
+    leads_values = list(CONFIG["leads"].values())
+    leads_keys = list(CONFIG["leads"])
+    start_date = pd.to_datetime(start_date)
+    leads_dict = {}
+    for idx, lead in enumerate(leads_keys):
+        target_range = predictions.target_range_format(leads_values[idx],leads_keys[idx],start_date,CONFIG["target_period_length"])
+        leads_dict.update({lead:target_range})
+    return leads_dict, leads_keys[0]
```

Comment:
you output leads_keys[0] here by in layout's leadTime, you point value directly to the CONFIG entry. You'd have to pick one or the other. I think pointing directly to the CONFIG file is good enough. If we need the callback to take of that at some point, we can revisit.